src/Castle.Windsor/MicroKernel/Lifestyle/Scoped/ThreadStaticLifetimeScope.cs

```diff
@@ -12,9 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#if !FEATURE_REMOTING && !FEATURE_ASYNCLOCAL
 namespace Castle.MicroKernel.Lifestyle.Scoped
 {
-#if SILVERLIGHT
```

Comment:
Go for it. My view is that we want conditionals to add features back in for old stuff (not yet or never going to be supported on the new runtime) rather than conditionals for what should be supported going forward.